core/embed/trezorhal/unix/touch/touch.c

```diff
@@ -31,17 +31,158 @@
 extern int sdl_display_res_x, sdl_display_res_y;
 extern int sdl_touch_offset_x, sdl_touch_offset_y;
 
-static bool _touch_detected = false;
 static int _touch_x = 0;
 static int _touch_y = 0;
 
+// distance from the edge where arrow button swipe starts [px]
+static const int _btn_swipe_begin = 20;
+// length of the arrow button swipe [px]
+static const int _btn_swipe_length = 60;
+
+// A state machine to handle both mouse inputs (simulating touch) and arrow
+// buttons (to simulate scroll movements). The variable `input_state` is used to
+// ensure that arrow keys are not processed when mouse input is in progress and
+// that mouse actions are not processed while arrow button swipe is not
+// finished.
+typedef enum {
+  IDLE,
```

Comment:
What about these names? They seem a bit clearer to me:

`MOUSE_DOWN_INSIDE`
`MOUSE_DOWN_OUTSIDE`

Reply:
refactored
https://github.com/trezor/trezor-firmware/pull/3619/commits/bedeb8a1be33d6fbc7c7ecb04adbfcab8514bb98